# Fix a crash in scgi_local or scgi_port when calling
# freeaddrinfo with a NULL pointer.
Index: rtorrent/src/command_network.cc
===================================================================
--- rtorrent/src/command_network.cc	(revision 1063)
+++ rtorrent/src/command_network.cc	(working copy)
@@ -261,10 +261,10 @@
       break;
     }
 
-    rak::address_info::free_address_info(ai);
+    if (ai != NULL) rak::address_info::free_address_info(ai);
 
   } catch (torrent::local_error& e) {
-    rak::address_info::free_address_info(ai);
+    if (ai != NULL) rak::address_info::free_address_info(ai);
 
     throw torrent::input_error(e.what());
   }
